gtsam/slam/tests/testSerializationDataset.cpp

```diff
@@ -0,0 +1,56 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file testSerializationDataset.cpp
+ * @brief serialization tests for dataset.cpp
+ * @author Ayush Baid
+ * @date Jan 1, 2021
+ */
+
+#include <gtsam/slam/dataset.h>
+
+#include <gtsam/base/serializationTestHelpers.h>
+#include <CppUnitLite/TestHarness.h>
+
+using namespace std;
+using namespace gtsam;
+using namespace gtsam::serializationTestHelpers;
+
+/* ************************************************************************* */
+TEST(dataSet, sfmDataSerialization) {
+  // Test the serialization of SfmData
+  const string filename = findExampleDataFile("dubrovnik-3-7-pre");
+  SfmData mydata;
+  CHECK(readBAL(filename, mydata));
+
+  EXPECT(equalsObj(mydata));
```

Comment:
very nice Ayush. Could you add an inline comment here explaining briefly what equalsObj, equalsXML, and equalsBinary are doing (round-trip equality check on serialization and subsequent deserialization) and why we want to check all 3 (obj, xml, and binary) representations?

Reply:
I do not know the specific reason for checking all three representations. I wrote all three because that is the way it has been done in other serialization unit tests.

Ref: https://github.com/borglab/gtsam/blob/fe53fa0818716cf674cf39cda7ebc3807d8eeb7f/gtsam/linear/tests/testSerializationLinear.cpp#L151

Reply:
Got it. Could we just add an inline comment for now above the `equalsObj()` lines explaining
```c++
// round-trip equality check on serialization and subsequent deserialization
```

@dellaert is there any specific reason to check 3 types (obj, xml, and binary) of representations in serialization throughout gtsam?

Reply:
Thanks. I have added it.

Reply:
thanks